### PR TITLE
feat: update gen ui section(s)

### PR DESCRIPTION
**Pull request by iteratetograceness**

### CODE REVIEW

File: content/docs/04-ai-sdk-rsc/04-generative-ui.mdx [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
+
+Built for use with React Server Components, `ai/rsc` is optimized for delivering highly-dynamic, personalized software like an e-commerce store, a web app, a social network -- and the most dynamic interfaces of them all: generative user interfaces. The benefits of using React Server Components for generative user interfaces include:
+
+- Reduced client bundle sizes
+- Fewer round trips between the server and the client
+- Improved developer experience around managing backend and frontend code
+
+<div className="mb-4">
+  <Note>
+    {' '}
```
Comment (iteratetograceness) [inline content/docs/04-ai-sdk-rsc/04-generative-ui.mdx:18]: question for reviewers: we should add a CTA here, where can we point users to for this callout?

Comment (lgrammel) [reply content/docs/04-ai-sdk-rsc/04-generative-ui.mdx:18]: good idea!

**Review by lgrammel — APPROVED** [review]

File: content/docs/04-ai-sdk-rsc/04-generative-ui.mdx (hunk 1 above)
Comment (iteratetograceness) [reply content/docs/04-ai-sdk-rsc/04-generative-ui.mdx:18]: jeremy is working on a full fledged example for this! will merge when ready, reverting this PR to draft

File: content/docs/06-advanced/02-why-generative-ui.mdx [hunk 2]
```diff
@@ -117,4 +124,4 @@ For example, when you're in a calendar application, you can ask the language mod
 
 Just by defining functions to lookup contacts, pull events from a calendar, and search for nearby locations, the model is able to sequentially navigate the routes for you.
 
-To learn more, check out these [examples](/examples/next-app/interface) using the `streamUI` function to stream generative user interfaces to the client based on the response from the language model.
+Learn how to create generative user interfaces with React Server Components [through `ai/rsc`](/examples/next-app/interface) or the framework-agnostic [`ai/ui` toolkit](/todo/jeremy/example).
```
Comment (iteratetograceness) [inline content/docs/06-advanced/02-why-generative-ui.mdx:127]: @jeremyphilemon TODO for me to link to your example!

File: content/docs/06-advanced/02-why-generative-ui.mdx [hunk 3]
```diff
@@ -1,17 +1,24 @@
...
-description: Overview of the AI and UI states
+title: Why Generative User Interfaces
+description: Why use generative user interfaces?
 ---
 
 # Generative User Interfaces
 
-Since language models can render user interfaces as part of their generations, the resulting model generations are referred to as generative user interfaces.
+Generative user interfaces elevate language models beyond plaintext and markdown. By rendering dynamic, adaptive user interfaces, language models can proactively adapt to the needs of the user.
+
```
Comment (iteratetograceness) [inline content/docs/06-advanced/02-why-generative-ui.mdx:9]: @nicoalbanese after moving this section to Advanced, I also removed the following copy, which I think we should move into the "Streaming React Components" section you're working on, wdyt?

```
Built for use with React Server Components, `ai/rsc` is optimized for delivering highly-dynamic, personalized software like an e-commerce store, a web app, a social network -- and the most dynamic interfaces of them all: generative user interfaces. The benefits of using React Server Components for generative user interfaces include:

- Reduced client bundle sizes
- Fewer round trips between the server and the client
- Improved developer experience around managing backend and frontend code
```

**Pull request closed without merging**